frontend/src/metabase/nav/hooks.unit.spec.tsx

```diff
@@ -0,0 +1,190 @@
+import { renderWithProviders, screen } from "__support__/ui";
+import { setupEnterpriseTest } from "__support__/enterprise";
+import { mockSettings } from "__support__/settings";
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import {
+  createMockDatabase,
+  createMockTokenFeatures,
+  createMockUser,
+} from "metabase-types/api/mocks";
+import { createMockState } from "metabase-types/store/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+
+import { useShouldShowDatabasePromptBanner } from "./hooks";
+
+interface setupOpts {
+  isAdmin?: boolean;
+  isPaidPlan?: boolean;
+  onlyHaveSampleDatabase?: boolean;
+  isOnAdminAddDatabasePage?: boolean;
+  isWhiteLabeling?: boolean;
+}
+const TEST_DB = createSampleDatabase();
+
+const DATA_WAREHOUSE_DB = createMockDatabase({ id: 2 });
+
+function setup({
+  isAdmin = false,
+  isPaidPlan = false,
+  isWhiteLabeling = false,
+  onlyHaveSampleDatabase = false,
+}: setupOpts = {}) {
+  if (onlyHaveSampleDatabase) {
+    setupDatabasesEndpoints([TEST_DB]);
+  } else {
+    setupDatabasesEndpoints([TEST_DB, DATA_WAREHOUSE_DB]);
+  }
+
+  const state = createMockState({
+    currentUser: createMockUser({ is_superuser: isAdmin }),
+    settings: mockSettings({
+      "token-features": createMockTokenFeatures(
+        isPaidPlan ? { sso: true } : {},
+      ),
+      "application-name": isWhiteLabeling ? "Acme Corp." : "Metabase",
+    }),
+  });
+
+  function TestComponent() {
+    const shouldShowDatabasePromptBanner = useShouldShowDatabasePromptBanner();
+    if (shouldShowDatabasePromptBanner == null) {
+      return <>Mock loading state</>;
+    }
+
+    return shouldShowDatabasePromptBanner ? (
+      <>showing database prompt banner</>
+    ) : (
+      <>hiding database prompt banner</>
+    );
+  }
+
+  renderWithProviders(<TestComponent />, {
+    storeInitialState: state,
+  });
+}
+
+describe("useShouldShowDatabasePromptBanner", () => {
+  beforeEach(() => {
+    setupEnterpriseTest();
+  });
+
+  it("should render for admin user with paid plan and has only sample database, but not white-labeled", async () => {
+    setup({
+      isAdmin: true,
+      isPaidPlan: true,
+      onlyHaveSampleDatabase: true,
+      isWhiteLabeling: false,
+    });
+
+    expect(
+      await screen.findByText("showing database prompt banner"),
+    ).toBeVisible();
+  });
+
+  it.each([
+    {
+      isPaidPlan: true,
```

Comment:
did you omit `true, true, false` by intention?

Reply:
Yeah, it's the case that actually renders the banner so it has its own `it()` block
https://github.com/metabase/metabase/blob/5fdb50be4ea717409c4c7d40a74b67b9933fd160/frontend/src/metabase/nav/hooks.unit.spec.tsx#L71-L77

Reply:
ah, now I see, thanks. it's pretty easy to get lost

Reply:
Yeah, it was never easy to structure tests, since there are so many conditions.

Reply:
I personally expect having a separate  describe block when you need to specify conditions, something like

```
describe("when admin user with paid plan and has only sample database, but not white-labeled", () => {
  it("should render a banner", () => {})
})
```

I'm not sure what @kamilmielnik or @oisincoveney think about it

Reply:
That also works. I think the different nuance is that
- `describe("when some conditions") -> it("should do something")` emphasizes the conditions while
- `it("should do something when conditions") emphasizes the result since it's positioned first.

I'm happy to change if we want one over another.

Reply:
I don't have strong opinion about it, I'm just sharing my thoughts after reading code. Maybe it's a quesiton to discuss together with the team and @ranquild when he returns as he is working on testing strategy atm 

so I'm not asking to change anything in scope of this PR, just a point to think about @WiNloSt